nixos/modules/services/finance/libeufin/module.nix

```diff
@@ -0,0 +1,29 @@
+{
+  lib,
+  pkgs,
+  config,
+  ...
+}:
+
+let
+  cfg = config.services.libeufin;
+  settingsFormat = pkgs.formats.ini { };
+in
+
+{
+  options.services.libeufin = {
+    configFile = lib.mkOption {
+      internal = true;
+      default = settingsFormat.generate "generated-libeufin.conf" cfg.settings;
+    };
```

Comment:
Commonly one would put this in a let-block.

```
let
  cfg = config.services.libeufin;
  format = pkgs.formats.ini { };
  configFile = format.generate "generated-libeufin.conf" cfg.settings;
in
```

Using the module system and `internal` to store variables like this is a pattern readers would immediately get very suspicious about.